examples/svdiff/README.md

```diff
@@ -0,0 +1,201 @@
+  --class_prompt="photo of a dog" \
+  --resolution=512 \
+  --train_batch_size=1 \
+  --gradient_accumulation_steps=1 \
+  --learning_rate=1e-3 \
+  --learning_rate_1d=1e-6 \
+  --train_text_encoder \
+  --lr_scheduler="constant" \
+  --lr_warmup_steps=0 \
+  --num_class_images=200 \
+  --max_train_steps=500
+```
+
+### Inference
+
+```python
+import torch
+from transformers import CLIPTextModel
+from diffusers import UNet2DConditionModel, StableDiffusionPipeline, DPMSolverMultistepScheduler
+
+from modeling_svdiff import set_spectral_shifts
+
+pretrained_model_name_or_path = "runwayml/stable-diffusion-v1-5"
+spectral_shifts_ckpt_dir = "ckpt-dir-path"
+device = "cuda" if torch.cuda.is_available() else "cpu"
+unet = UNet2DConditionModel.from_pretrained(pretrained_model_name_or_path, subfolder="unet").to(device)
+text_encoder = CLIPTextModel.from_pretrained(pretrained_model_name_or_path, subfolder="text_encoder").to(device)
+# make sure to move models to device before
+unet, _ = set_spectral_shifts(unet, spectral_shifts_ckpt=os.path.join(spectral_shifts_ckpt, "spectral_shifts.safetensors"))
+text_encoder, _ = set_spectral_shifts(text_encoder, spectral_shifts_ckpt=os.path.join(spectral_shifts_ckpt, "spectral_shifts_te.safetensors"))
+# load pipe
+pipe = StableDiffusionPipeline.from_pretrained(
+    model_id,
+    unet=unet,
+    text_encoder=text_encoder,
+)
+pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
+pipe.to(device)
+image = pipe("A picture of a sks dog in a bucket", num_inference_steps=25).images[0]
+```
+
+## Single Image Editing
+### Training
+In Single Image Editing, your instance prompt should be just the description of your input image **without the identifier**. 
+
+```bash
+export MODEL_NAME="runwayml/stable-diffusion-v1-5"
+export INSTANCE_DIR="dir-path-to-input-image"
+export CLASS_DIR="path-to-class-images"
+export OUTPUT_DIR="path-to-save-model"
+
+accelerate launch train_svdiff.py \
+  --pretrained_model_name_or_path=$MODEL_NAME  \
+  --instance_data_dir=$INSTANCE_DIR \
+  --class_data_dir=$CLASS_DIR \
+  --output_dir=$OUTPUT_DIR \
+  --instance_prompt="photo of a pink chair with black legs" \
+  --resolution=512 \
+  --train_batch_size=1 \
+  --gradient_accumulation_steps=1 \
+  --learning_rate=1e-3 \
+  --learning_rate_1d=1e-6 \
+  --train_text_encoder \
+  --lr_scheduler="constant" \
+  --lr_warmup_steps=0 \
+  --max_train_steps=500
+```
+
+### Inference
+
+```python
+from PIL import Image
+import torch
+from transformers import CLIPTextModel
+from diffusers import UNet2DConditionModel, DDIMScheduler
+
+from modeling_svdiff import set_spectral_shifts
+from pipeline_stable_diffusion_ddim_inversion import StableDiffusionPipelineWithDDIMInversion
+
+pretrained_model_name_or_path = "runwayml/stable-diffusion-v1-5"
+spectral_shifts_ckpt_dir = "ckpt-dir-path"
+image = "path-to-image"
+source_prompt = "prompt-for-image"
+target_prompt = "prompt-you-want-to-generate"
```

Comment:
I'd prefer a more complete example here with respect to the `instance_prompt` shown just above.

Reply:
`instance_prompt` is more like a prompt for the training image.

Reply:
I guess both are good, but for single-image editing, using `source_prompt` and `target_prompt` appears to be more conventional, like here:
https://github.com/TencentARC/MasaCtrl/blob/main/playground_real.ipynb